### PR TITLE
LF-3932: Address comments in #3055

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

Address comments in https://github.com/LiteFarmOrg/LiteFarm/pull/3055
- wrap the logo with `TextButton`
- rename a prop 

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ import {
 import Alert from '../../../containers/Navigation/Alert';
 import { useTranslation } from 'react-i18next';
 import clsx from 'clsx';
+import TextButton from '../../Form/Button/TextButton';
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx:33]: Just wondering if we should continue using our custom TextButton? Or use one of MUI's buttons like IconButton or something moving forward.

File: packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx [hunk 2]
```diff
@@ -221,19 +222,15 @@ const TopMenu = ({ history, isMobile, showNavigation, onClickBurger }) => {
...
-        {...props}
-      />
-    ) : (
-      <WordsLogo
-        alt="LiteFarm Logo"
-        className={clsx(styles.logo, withLink && styles.withLink, styles.paddingTopBottom)}
-        {...props}
-      />
+  const Logo = ({ withoutWords, onClick }) => {
+    if (!withoutWords) {
```
Comment (antsgar) [inline packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx:226]: The double negation is a bit confusing here, could we have this prop be `withWords` instead, and then we could just do `if (withWords) {...`?

Alternatively we could do `if (withoutWords)` as the first block here.

File: packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx]: It felt awkward to me too that we were using TextButton for an image. IconButton seems good, thank you!

File: packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx (hunk 2 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Navigation/TopMenu/TopMenu.jsx]: I confused myself too... I went with the second solution, thank you!

**Review by antsgar — APPROVED** [review]

**Review by Duncan-Brain — APPROVED** [review]
Looks good!

**Pull request merged**